contracts/RootChain.sol

```diff
@@ -135,15 +147,22 @@ contract RootChain {
         // Only allow up to CHILD_BLOCK_INTERVAL deposits per child block.
         require(currentDepositBlock < CHILD_BLOCK_INTERVAL);
 
-        bytes32 root = keccak256(msg.sender, address(0), msg.value);
-        uint256 depositBlock = getDepositBlock();
-        childChain[depositBlock] = ChildBlock({
-            root: root,
-            timestamp: block.timestamp
-        });
-        currentDepositBlock = currentDepositBlock.add(1);
+        emitDepositBlock(msg.sender, address(0), msg.value);
+    }
+
 
-        emit Deposit(msg.sender, depositBlock, address(0), msg.value);
+    /**
+     * @dev Deposits approved amount of ERC20 token. Approve must be called first. Note: does not check if token was added.
```

Comment:
Checking the token was added seems important. Was there a specific reason not to?

Reply:
Token must be added before any exists are being made, otherwise `addExitToQueue` will throw. Checking if token was added here incures additional gas cost, since we can't really remove check from `addExitToQueue`. So question is - does it provide additional safety? I can't find good arguments for that.